### ISSUE TITLE
Exception when trying to load exchange AllCoin , Bibox ,Brazilex ,Coinegg ,OkCoinUSD 

### ISSUE THREAD

**Issue by DaveMG3** [ccxt/ccxt#2880]
OS: Windows
Programming Language version: Php
CCXT version: 1.14.10

Hello Kroitor,
I tested the last version of CCTX and noticed some 503 errors on some exchange links (which are fine when accessed from browser).
Is it a known issue or a misuse from my side?

AllCoin : Exception when trying to load exchange: allcoin GET https://www.allcoin.com/Home/MarketOverViewDetail/ 51 SSL: no alternative certificate subject name matches target host name 'www.allcoin.com'

Bibox : Exception when trying to load exchange: bibox GET https://api.bibox.com/v1/mdata?cmd=marketAll 503 

Brazilex : Exception when trying to load exchange: braziliex GET https://braziliex.com/api/v1/public/ticker 503 not accessible from this location at the moment

Coinegg : Exception when trying to load exchange: coinegg GET https://www.coinegg.com/coin/btc/allcoin?t=1526584832951 503 not accessible from this location at the moment

Ice3x : Exception when trying to load exchange: ice3x GET https://ice3x.com/api/v1/currency/list 403 not accessible from this location at the moment

OkCoinUSD : Exception when trying to load exchange: okcoinusd GET https://www.okcoin.com/v2/markets/products.do 28 Operation timed out after 10015 milliseconds with 0 bytes received

**Comment by kroitor** [comment]
> AllCoin : Exception when trying to load exchange: allcoin GET https://www.allcoin.com/Home/MarketOverViewDetail/ 51 SSL: no alternative certificate subject name matches target host name 'www.allcoin.com'

They updated their API and their website, so the support for AllCoin is currently broken in CCXT.

> Bibox : Exception when trying to load exchange: bibox GET https://api.bibox.com/v1/mdata?cmd=marketAll 503

503 can indicate a problem on their side or a crude ddos protection of some sort, try accessing https://api.bibox.com/v1/mdata?cmd=marketAll from your browser and refresh the page a few times, then make a pause, then refresh a few times again and see if it is repeatable without CCXT in general. Basically, that particular endpoint doesn't seem to respond well. Will see what we can do about it.

> Coinegg : Exception when trying to load exchange: coinegg GET https://www.coinegg.com/coin/btc/allcoin?t=1526584832951 503 not accessible from this location at the moment

The problem with Coinegg is that they don't have an endpoint for a list of coins documented anywhere in their API docs: https://www.coinegg.com/explain.api.html
So, the only way to get some pairs from them was to load their website data, otherwise we would have to hardcode the markets for this exchange, which is not something we would want to do.
Loading from Web means requesting these URLs:

https://www.coinegg.com/coin/btc/allcoin?t=1526589506859
https://www.coinegg.com/coin/eth/allcoin?t=1526589616337
https://www.coinegg.com/coin/usc/allcoin?t=1526589739777

↑ One of three usually fails randomly and silently (this is in fact a Cloudflare protection error).

So the proper way to access it is to set a greater rateLimit and use a cloudscraper or scrape cloudflare cookies and set proper headers, like:

```
{ Cookie: 'cf_clearance=ec60d1bf1adc47e1e2f4322aeecce2aecd54d9cd-1526591097-900; __cfduid=d24943438f0bcc70693f241718ded3b291526591088',
  'User-Agent': 'Mozilla/5.0 (Windows NT 6.1; Win64; x64; rv:60.0) Gecko/20100101 Firefox/60.0' }
```

> Brazilex : Exception when trying to load exchange: braziliex GET https://braziliex.com/api/v1/public/ticker 503 not accessible from this location at the moment

Same Cloudflare protection as with Coinegg.

> Ice3x : Exception when trying to load exchange: ice3x GET https://ice3x.com/api/v1/currency/list 403 not accessible from this location at the moment

Same thing as with previous two.

> OkCoinUSD : Exception when trying to load exchange: okcoinusd GET https://www.okcoin.com/v2/markets/products.do 28 Operation timed out after 10015 milliseconds with 0 bytes received

Fixed in #2863 and will be available in version 1.14.13 in 5-10 minutes.

Hope this answers your questions. Let us know if not.

**Comment by DaveMG3** [comment]
Yep all answered. Thanks for your prompt feedback. Guess i need to learn a little more on cloud scrapper.

I retested with 1.14.20 and noticed the 3 below failing too :

ACX : (similar to above fixed OkCoin) Exception when trying to load exchange: acx GET https://acx.io/api/api/v2/markets.json 28 Operation timed out after 10016 milliseconds with 0 out of 0 bytes received

KuCoin: (all fine with browser) Exception when trying to load exchange: kucoin GET https://api.kucoin.com/v1/market/open/symbols 404 not accessible from this location at the moment

YunBi : (503 also from browser) Exception when trying to load exchange: yunbi GET https://yunbi.com/api/v2/markets.json **503** <html>
<head><title>503 Service Temporarily Unavailable</title></head>
<body bgcolor="white">
<center><h1>503 Service Temporarily Unavailable</h1></center>
<hr><center>nginx/1.6.2</center>
</body>
</html>
